### PR TITLE
Move COM basics to Core & convert more GDI+ interop

### PR DESCRIPTION
**Pull request by JeremyKuhne**
This change converts the Image and Bitmap classes to CsWin32.

- Move IStream, ComHelpers, IDispatch, VARIANT and other core COM code to Core
- Change the name of GpStream to ComManagedStream and consolidate both versions
- Create an extension method for Stream to create an IStream scope
- Take in the latest CsWin32 drop to fix a blocking COM generation issue
- Use the new option to not generate friendly overloads in System.Drawing (there is a pending metadata issue that causes some of these to be generated unsafely).
- Update some code to throw ArgumentNull instead of NullRef
- Fix PrinterSettings.InstalledPrinters and add a test

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10667)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
Fixes #10659

**Comment by elachlan** [comment]
```
src\System.Drawing.Common\src\System\Drawing\Image.cs(11,1): error IDE0005: (NETCORE_ENGINEERING_TELEMETRY=Build) Using directive is unnecessary. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0005)
src\System.Drawing.Common\src\System\Drawing\Printing\PrinterSettings.cs(23,23): error CA1823: (NETCORE_ENGINEERING_TELEMETRY=Build) Unused field 'Padding64Bit' (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1823)

```

**Comment by elachlan** [comment]
This might also fix #10668

**Comment by JeremyKuhne** [comment]
> This might also fix #10668

Yes, it should. We were trying to print to an information context. Hopefully I've shaken out the rest of the test failures.

File: src/System.Drawing.Common/src/System/Drawing/Bitmap.cs [hunk 1]
```diff
@@ -259,42 +239,55 @@ public void MakeTransparent(Color transparentColor)
...
-            new HandleRef(this, _nativeImage), ref rect, flags, format, bitmapData);
+        ArgumentNullException.ThrowIfNull(bitmapData);
+
+        Rect nativeRect = rect;
 
-        // libgdiplus has the wrong error code mapping for this state.
-        if (status == 7)
+        fixed (void* data = &bitmapData.GetPinnableReference())
         {
-            status = 8;
```
Comment (lonitra) [inline src/System.Drawing.Common/src/System/Drawing/Bitmap.cs:286]: Is this issue still present with current code?

File: src/System.Private.Windows.Core/src/Windows/Win32/System/Com/Interop.GPStream.cs [hunk 2]
```diff
@@ -0,0 +1,248 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (lonitra) [inline src/System.Private.Windows.Core/src/Windows/Win32/System/Com/Interop.GPStream.cs:1]: Should this file be renamed too?

File: src/System.Drawing.Common/src/System/Drawing/Image.cs [hunk 3]
```diff
@@ -405,32 +364,28 @@ public void Save(Stream stream, ImageCodecInfo encoder, EncoderParameters? encod
...
             }
+
+            GC.KeepAlive(this);
+            GC.KeepAlive(encoderParams);
         }
     }
 
     /// <summary>
-    ///  Adds an <see cref='EncoderParameters'/> to this <see cref='Image'/>.
+    ///  Adds an <see cref='Imagin.EncoderParameters'/> to this <see cref='Image'/>.
```
Comment (lonitra) [inline src/System.Drawing.Common/src/System/Drawing/Image.cs:384]:
```suggestion
    ///  Adds an <see cref='Imaging.EncoderParameters'/> to this <see cref='Image'/>.
```

File: src/System.Private.Windows.Core/src/Windows/Win32/System/Com/Interop.GPStream.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Private.Windows.Core/src/Windows/Win32/System/Com/Interop.GPStream.cs:1]: Yeah, in a follow up. Don't want to mess up history.

File: src/System.Drawing.Common/src/System/Drawing/Bitmap.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Drawing.Common/src/System/Drawing/Bitmap.cs:286]: libgdiplus is the Unix implementation of GDI+ functionality, so no

**Review by lonitra — APPROVED** [review]
:shipit:

**Pull request merged**